### PR TITLE
Query search fix/yaf 105

### PR DESCRIPTION
**Pull request by mvolonnino**

### CODE REVIEW

File: functions/controllers/funkoPopController.js [hunk 1]
```diff
@@ -42,24 +55,40 @@ const getFunkoPopName = async (req, res, next) => {
...
       funkoArray.forEach((funko) => {
+        const genre = funko.genre;
         const funkoData = funko.funkoData;
-        const matches = funkoData.filter((data) =>
-          data.name.toLowerCase().includes(query)
-        );
-        console.log(matches);
+        console.log("funkoData: ", funkoData);
+        if (funkoData) {
+          if (funkoData.name) {
```
Comment (cvols) [inline functions/controllers/funkoPopController.js:73]: we don't need 2 if checks here.  just if (funkoData?.name) or (funkoData && funkoData.name)

File: functions/controllers/funkoPopController.js [hunk 2]
```diff
@@ -5,26 +5,40 @@ const firestore = firebase.firestore();
...
+      let funkoData = data.data().funkoData;
+      dataToAdd.funkoData.forEach((funko) => {
+        funkoData.push(funko);
+      });
+      const funkoAdd = {
+        genre: dataToAdd.genre,
+        funkoData: funkoData,
+      };
+      await genreCheck.update(funkoAdd);
+      res.send("Record added successfully");
```
Comment (cvols) [inline functions/controllers/funkoPopController.js:21]: this needs to send back a status

File: functions/controllers/funkoPopController.js [hunk 3]
```diff
@@ -5,26 +5,40 @@ const firestore = firebase.firestore();
...
+    } else {
+      await genreCheck.set(dataToAdd);
+      res.send("Record created successfully");
```
Comment (cvols) [inline functions/controllers/funkoPopController.js:24]: this needs to send back a status

File: src/App.js [hunk 4]
```diff
@@ -8,15 +8,21 @@ import "./App.css";
...
 
   useEffect(() => {
     auth.onAuthStateChanged((authUser) => {
       if (authUser) {
         if (!user) {
           dispatch({
             type: "SET_USER",
-            user: authUser.providerData[0],
+            user: {
+              uid: authUser.uid,
```
Comment (cvols) [inline src/App.js:19]: you should deconstruct uid, displayName, email etc off authUser so you dont have object lookups every time.

const { uid, displayName } = authUser;

File: src/pages/HomePage/components/Home/Home.jsx [hunk 5]
```diff
@@ -1,14 +1,26 @@
...
-import './Home.css';
+import "./Home.css";
 
 function Home() {
   const [{ user }, dispatch] = useDataLayerValue();
 
+  useEffect(() => {
+    API.getFunkoPopData()
+      .then((res) => {
+        const data = res.data;
```
Comment (cvols) [inline src/pages/HomePage/components/Home/Home.jsx:16]: this should be const { data } = res

File: src/pages/LoginPage/components/Login/Login.jsx [hunk 6]
```diff
@@ -1,72 +1,124 @@
...
-      .then(user => {
+      .then((user) => {
         // signed up
         var u = auth.currentUser;
         u.updateProfile({
-          displayName: userObj.displayName
+          displayName: userObj.displayName,
         })
           .then(() => {
+            API.signUpUser({ u })
```
Comment (cvols) [inline src/pages/LoginPage/components/Login/Login.jsx:41]: what does u mean here? you should use full names so other people know what's going on

File: src/pages/LoginPage/components/Login/Login.jsx [hunk 7]
```diff
@@ -1,72 +1,124 @@
...
+              })
+              .catch((error) => {
+                console.error(error);
+              });
             dispatch({
-              type: 'SET_USER',
-              user: u.providerData[0]
+              type: "SET_USER",
+              user: {
+                uid: u.uid,
```
Comment (cvols) [inline src/pages/LoginPage/components/Login/Login.jsx:51]: deconstruct uid, displayName etc here as well

File: src/pages/LoginPage/components/Login/Login.jsx [hunk 8]
```diff
@@ -1,72 +1,124 @@
...
+  const handleAuthLogin = (e) => {
     e.preventDefault();
     switch (e.currentTarget.value) {
-      case 'google':
-        auth.signInWithPopup(gProvider).catch(err => alert(err.message));
+      case "google":
+        auth
+          .signInWithPopup(gProvider)
+          .then(() => {
+            const u = auth.currentUser;
```
Comment (cvols) [inline src/pages/LoginPage/components/Login/Login.jsx:97]: should be const { currentUser } = auth

File: src/utils/API.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+
+export default {
+  signUpUser: function ({ u }) {
```
Comment (cvols) [inline src/utils/API.js:4]: can we change u to user?

File: src/utils/API.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    return axios({
+      method: "POST",
+      data: {
+        uid: u.uid,
+        displayName: u.displayName,
```
Comment (cvols) [inline src/utils/API.js:9]: should deconstruct here as well

File: functions/controllers/funkoPopController.js (hunk 1 above)
Comment (mvolonnino) [reply functions/controllers/funkoPopController.js:73]: not using this function anymore, have it just for reference but i will change it

File: src/App.js (hunk 4 above)
Comment (mvolonnino) [reply src/App.js:19]: done

File: src/pages/HomePage/components/Home/Home.jsx (hunk 5 above)
Comment (mvolonnino) [reply src/pages/HomePage/components/Home/Home.jsx:16]: done

File: src/pages/LoginPage/components/Login/Login.jsx (hunk 6 above)
Comment (mvolonnino) [reply src/pages/LoginPage/components/Login/Login.jsx:41]: could i deconstruct here? so const {currentUser} = auth; ?

Comment (mvolonnino) [reply src/pages/LoginPage/components/Login/Login.jsx:41]: done

File: src/pages/LoginPage/components/Login/Login.jsx (hunk 7 above)
Comment (mvolonnino) [reply src/pages/LoginPage/components/Login/Login.jsx:51]: done

File: src/pages/LoginPage/components/Login/Login.jsx (hunk 8 above)
Comment (mvolonnino) [reply src/pages/LoginPage/components/Login/Login.jsx:97]: done

File: src/utils/API.js (hunk 9 above)
Comment (mvolonnino) [reply src/utils/API.js:4]: changed it to the deconstructed variables being passed, so now its like
signUpUser function({uid, displayName, etc})

File: src/utils/API.js (hunk 10 above)
Comment (mvolonnino) [reply src/utils/API.js:9]: done

File: functions/controllers/funkoPopController.js (hunk 2 above)
Comment (mvolonnino) [reply functions/controllers/funkoPopController.js:21]: done

File: functions/controllers/funkoPopController.js (hunk 3 above)
Comment (mvolonnino) [reply functions/controllers/funkoPopController.js:24]: done

**Pull request merged**